crates/extensions/az_mapper_ext/src/converter.rs

```diff
@@ -48,24 +54,89 @@ impl AzureConverter {
     }
 
     fn try_convert(&mut self, input: &MqttMessage) -> Result<Vec<MqttMessage>, ConversionError> {
-        let default_timestamp = self.add_timestamp.then(|| self.clock.now());
+        if is_bridge_health(&input.topic.name) {
+            Ok(vec![])
+        } else {
+            match self.mqtt_schema.entity_channel_of(&input.topic) {
+                Ok((source, channel)) => self.try_convert_te_topics(source, channel, input),
+                Err(_) => self.try_convert_tedge_health_topics(input),
+            }
+        }
+    }
+
+    fn try_convert_tedge_health_topics(
+        &mut self,
+        input: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
         if is_bridge_health(&input.topic.name) {
             Ok(vec![])
         } else if input.topic.name.starts_with("tedge/health") {
             let mut input_msg = input.clone();
-            input_msg.topic = self.mapper_config.out_topic.clone();
+            input_msg.topic = Topic::new_unchecked(&self.mapper_config.out_topic.clone());
             Ok(vec![input_msg])
         } else {
-            self.size_threshold.validate(input)?;
+            Ok(vec![])
+        }
+    }
+
+    fn try_convert_te_topics(
+        &mut self,
+        source: EntityTopicId,
+        channel: Channel,
+        input: &MqttMessage,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        let mea_type = match &channel {
+            Channel::Measurement { measurement_type } => measurement_type,
+            Channel::Event { event_type } => event_type,
+            Channel::Alarm { alarm_type } => alarm_type,
+            _ => return Ok(vec![]),
+        };
+        self.convert_message(input, source, mea_type)
+    }
+
+    fn convert_message(
+        &mut self,
+        input: &MqttMessage,
+        source: EntityTopicId,
+        telemetry_type: &String,
+    ) -> Result<Vec<MqttMessage>, ConversionError> {
+        let payload = self.with_timestamp(input)?;
+        let source = normalize_name(source);
+
+        let out_topic = match input.topic.name.split('/').collect::<Vec<_>>()[..] {
+            [_, _, _, _, _, "m", _] => Topic::new_unchecked(&format!(
+                "{0}{source}/m/{telemetry_type}",
```

Comment:
I feel that we are safe here because not many were using the older version. @reubenmiller can clarify this.

Reply:
I don't think Azure support/recommends publishing to nested topics on the `az/messages/events` topic. I think the idea is to use the property bags instead (as Az prefers to use a generic translation layer) rather than just relying on MQTT topic semantics (but I'm still trying to find relevant documentation on their website)

Reply:
Here is the Azure docs
https://learn.microsoft.com/en-us/azure/iot/iot-mqtt-connect-to-iot-hub#sending-device-to-cloud-messages

In summary it says that publishing data from a device to the cloud should be published in the following format:

```
devices/{device-id}/messages/events/ or devices/{device-id}/messages/events/{property-bag}
```

Where the property bag should be URL encoded, e.g.

```
devices/{device-id}/messages/events/$.ct=application%2Fjson%3Bcharset%3Dutf-8
```

So we can't just blindly forward our topic structure, as that will properly result in the messages just being ignored/lost...or even if it works today for some reason, then we run the risk that it will stop working some time in the future as it is out-of-spec.

Reply:
But using property bags should not be breaking changes on the Azure IoT Hub side, so we don't have to worry about that.

Reply:
So, I feel we can use the below topic format.

devices/{device-id}/messages/events/telemetry_type={M/E/A}%3Btypename="type"
{device-id} is anyways is derived using the `te` topics
For the main device, device-id will be => device:main
For the child device device-id will be =>device:main:child
For the main device service, it will be => device:main:service:service-id
For the child device service, it will be => device:main:child:service:service-id

Reply:
Created a ticket to address this issue in a followup ticket https://github.com/thin-edge/thin-edge.io/issues/2317